internal/shortener/handlers/handlers.go

```diff
@@ -230,3 +251,115 @@ func (h *handlers) Ping(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func (h *handlers) getUserCode(w http.ResponseWriter, r *http.Request) (string, error) {
+
+	// куки пользователя
+	var userCode string
+	tokenCookie, err := r.Cookie(cCookieUser)
+	if err != nil {
+		userCode = h.shortener.GetNewUserCode()
+		tokenString, err := token.BuildJWTString(userCode)
+		if err != nil {
+			return "", err
+		}
+		tokenCookie := http.Cookie{
+			Name:  cCookieUser,
+			Value: tokenString,
+		}
+		http.SetCookie(w, &tokenCookie)
+	} else {
+		userCode, err = token.GetUserCode(tokenCookie.Value)
+		if err != nil {
+			return "", err
+		}
+	}
+	return userCode, nil
+}
+
+func (h *handlers) getUserCodeReadOnly(r *http.Request) (string, error) {
+
+	// куки пользователя
+	var userCode string
+	tokenCookie, err := r.Cookie(cCookieUser)
+	if err != nil {
+		return "", err
+	} else {
+		userCode, err = token.GetUserCode(tokenCookie.Value)
+		if err != nil {
+			return "", err
+		}
+	}
+	return userCode, nil
+}
+
+type GetUserURLsJSON struct {
+	ShortURL    string `json:"short_url"`
+	OriginalURL string `json:"original_url"`
+}
+
+func (h *handlers) GetUserURLs(w http.ResponseWriter, r *http.Request) {
+	userCode, err := h.getUserCodeReadOnly(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	batch, err := h.shortener.GetShortnerBatchUser(userCode)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var response []GetUserURLsJSON
+	for _, row := range batch {
+		shortURL := fmt.Sprintf("http://%s/%s", h.baseaddr, row.Key.Code)
+		response = append(response, GetUserURLsJSON{ShortURL: shortURL, OriginalURL: row.Data.URL})
+	}
+
+	if len(response) > 0 {
+		responseJSON, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(responseJSON)
+	} else {
+		w.WriteHeader(http.StatusNoContent)
+	}
+}
+
+func (h *handlers) DeleteShortenerBatch(w http.ResponseWriter, r *http.Request) {
+	// получение id пользователя
+	userCode, err := h.getUserCodeReadOnly(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	// Чтение body
+	var buf bytes.Buffer
+	_, err = buf.ReadFrom(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	var codeArr []string
+	err = json.Unmarshal(buf.Bytes(), &codeArr)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Конвертация
+	var s []model.Shortener
```

Comment:
```suggestion
	s := make([]model.Shortener, 0, len(codeArr))
```
https://t.me/bushigo/12